packages/expo-sharing/ios/EXSharing/EXSharingModule.m

```diff
@@ -68,7 +67,19 @@ - (void)documentInteractionControllerDidDismissOpenInMenu:(UIDocumentInteraction
   _pendingResolver(nil);
   _pendingResolver = nil;
 
-  _documentInteractionController = nil;
+  // This delegate method is called whenever:
+  // a) the share sheet is cancelled
+  // b) an app is chosen, it's dialog opened, and the share is confirmed/ sent
+  // c) an app is chosen, it's dialog opened, and that dialog is cancelled
+  // In case c), the share sheet remains open, even though the promise was resolved
+  // Future attempts to share without closing the sheet will fail to attach the file, so we need to close it
+  // No other delegate methods fire only when the share sheet is dismissed, unfortunately
+  EX_WEAKIFY(self);
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
I'm wondering if this is really necessary, I would say that those delegate methods are already called on the main thread since it interacts with the UI. Can you verify that?

Reply:
This is rather strange! This delegate method is indeed called on the main thread (or, so says the debugger), but the share sheet fails to close (silently; no console warnings, etc.) unless it's inside the `dispatch_async`... so, this appears to be necessary.

Since clearly the thread isn't the problem, I suppose it has something to do with the call being async. Could something be preventing the `UIDocumentInteractionController` from dismissing itself directly inside its own delegate's method? ¯\_(ツ)_/¯